sdcm/cassandra_harry_thread.py

```diff
@@ -50,118 +50,52 @@ def run(self):
-                    ret.append(node_cs_res)
-            except Exception as exc:  # pylint: disable=broad-except
-                CassandraHarryEvent.error(node="",
-                                          stress_cmd=self.stress_cmd,
-                                          errors=[f"Failed to proccess stress summary due to {exc}"]).publish()
+    def __init__(self, *args, **kwargs):
+        credentials = kwargs.pop('credentials', None)
 
-        return ret
-
-    def verify_results(self):
-        harry_summary = []
-        results = []
-        errors = []
-
-        LOGGER.debug('Wait for stress threads results')
-        for future in concurrent.futures.as_completed(self.results_futures, timeout=self.timeout):
-            results.append(future.result())
-
-        for node, result in results:
-            if not result:
-                # Silently skip if stress command throwed error, since it was already reported in _run_stress
-                continue
-            output = result.stdout + result.stderr
+        super().__init__(*args, **kwargs)
 
-            lines = output.splitlines()
-            node_cs_res = self._parse_harry_summary(lines)  # pylint: disable=protected-access
+        if credentials and "username=" not in self.stress_cmd:
+            self.stress_cmd += " -username {} -password {}".format(*credentials)
 
-            if node_cs_res:
-                harry_summary.append(node_cs_res)
+    def _run_stress(self, loader, loader_idx, cpu_idx):
+        # Select first seed node to send the scylla-harry cmds
+        ip = self.node_list[0].cql_ip_address
 
-            for line in lines:
-                if 'java.io.IOException' in line:
-                    errors += ['%s: %s' % (node, line.strip())]
+        if not os.path.exists(loader.logdir):
+            os.makedirs(loader.logdir, exist_ok=True)
+        log_file_name = os.path.join(loader.logdir, f'cassandra-harry-l{loader_idx}-{uuid.uuid4()}.log')
+        LOGGER.debug('cassandra-harry local log: %s', log_file_name)
 
-        return harry_summary, errors
+        LOGGER.debug("running: %s", self.stress_cmd)
 
-    def run(self):
-        if self.round_robin:
-            loaders = [self.loader_set.get_loader()]
+        if self.stress_num > 1:
+            node_cmd = f'taskset -c {cpu_idx} bash -c "{self.stress_cmd}"'
         else:
-            loaders = self.loader_set.nodes if not self.use_single_loader else [self.loader_set.nodes[0]]
-        LOGGER.debug("Round-Robin through loaders, Selected loader is {} ".format(loaders))
+            node_cmd = self.stress_cmd
 
-        self.max_workers = (os.cpu_count() or 1) * 5
-        LOGGER.debug("Starting %d cassandra-harry Worker threads", self.max_workers)
-        # pylint: disable=consider-using-with
-        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=self.max_workers)
+        docker = RemoteDocker(loader, 'scylladb/hydra-loaders:cassandra-harry-jdk11-20220816',
```

Comment:
how about keeping image name/tag in config defaults?

Reply:
it would, the change setting up the configuration for it is in a different PR,

this can be merged like this (BTW, the ycsb and some other docker loaders, are hardcoded the same)